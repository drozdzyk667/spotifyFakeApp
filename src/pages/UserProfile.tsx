import React from "react";

const style = {
  container: {
    display: "flex",
    flexDirection: "row",
    padding: "2em",
    margin: "2em"
  } as React.CSSProperties,
  spacer: {
    margin: "2em"
  }
};

const UserProfile = () => {
  const [isLoading, setIsLoading] = React.useState(false);
  const [error, setError] = React.useState();
  const [capturedData, setCapturedData] = React.useState();
  const USER_URL = "https://api.spotify.com/v1/me";

  const getUserInfo = async () => {
    setIsLoading(true);
    await fetch(USER_URL, {
      method: "GET",
      headers: {
        Authorization: "Bearer " + localStorage.getItem("accessToken")
      }
    })
      .then(res => {
        return res.ok ? res.json() : null;
      })
      .then(data => {
        setCapturedData(data);
        console.log(data);
        setIsLoading(false);
      })
      .catch(error => {
        setError(error);
      });
  };

  React.useEffect(() => {
    getUserInfo();
  }, []);

  if (error) {
    return <p> {error?.message}</p>;
  }

  if (isLoading) {
    return <p> {"Loading User data..."}</p>;
  }

  return (
    <div style={style.container}>
      <div>
        <img
          style={{ borderRadius: "50%" }}
          src={capturedData?.images[0].url}
          alt="user_photo"
        />
      </div>
      <div style={style.spacer}>
        <h2>{capturedData?.display_name.slice(0, 5)}</h2>
        <h4>{capturedData?.email}</h4>
      </div>
    </div>
  );
};

export default UserProfile;
