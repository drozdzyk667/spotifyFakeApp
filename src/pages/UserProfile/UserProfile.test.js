import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from './UserProfile';

const clickFn = jest.fn();
describe('User Profile', () => {
  it('button click should open external spotify user account page', () => {
    const component = shallow(<UserProfile onClick={clickFn} />);
    expect(
      component.find('a').text('Visit Your Spotify Profile'),
    ).toMatchSnapshot();
    expect(
      component
        .find('a')
        .findWhere(d => d.prop('data-testid'))
        .simulate('click', true),
    ).toMatchSnapshot();
  });
});
