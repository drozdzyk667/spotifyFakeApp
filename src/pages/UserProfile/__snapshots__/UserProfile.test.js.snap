// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile button click should open external spotify user account page 1`] = `"Visit Your Spotify Profile"`;

exports[`User Profile button click should open external spotify user account page 2`] = `
<a
  data-testid="userProfile-button"
  rel="noopener noreferrer"
  style={
    Object {
      "backgroundColor": "black",
      "borderRadius": "10px",
      "color": "white",
      "padding": "0.5em",
      "textDecoration": "none",
    }
  }
  target="_blank"
>
  Visit Your Spotify Profile
</a>
`;
